packages/nimble-components/src/table-column/icon/tests/table-column-icon.spec.ts

```diff
@@ -0,0 +1,396 @@
+        const source = new Model();
+        const result = await fixture<Table<SimpleTableRecord>>(html<Model>`
+            <${tableTag} style="width: 700px">
+                <${tableColumnIconTag} ${ref('col1')} field-name="field1" key-type="${keyType}">
+                    Column 1
+                    ${repeat(() => mappings, html<BasicIconMapping>`
+                        <${mappingIconTag}
+                            key="${x => x.key}"
+                            text="${x => x.text}"
+                            icon="${x => x.icon}">
+                        </${mappingIconTag}>
+                    `)}
+                </${tableColumnIconTag}>
+            </${tableTag}>`, { source });
+        return {
+            ...result,
+            model: source
+        };
+    }
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    for (const test of [
+        { type: 'string', key: 'a' },
+        { type: 'number', key: 10 },
+        { type: 'boolean', key: true }
+    ]) {
+        // eslint-disable-next-line @typescript-eslint/no-loop-func
+        it(`displays icon mapped from ${test.type}`, async () => {
+            ({ element, connect, disconnect, model } = await setup(
+                [{ key: test.key, text: 'alpha', icon: iconXmarkTag }],
+                test.type
+            ));
+            pageObject = new TablePageObject<SimpleTableRecord>(element);
+            await element.setData([{ field1: test.key }]);
+            await connect();
+            await waitForUpdatesAsync();
+
+            expect(
+                pageObject.getRenderedCellIcon(0, 0) instanceof IconXmark
+            ).toBeTrue();
+        });
+    }
+
+    it('displays blank when no matches', async () => {
+        ({ element, connect, disconnect, model } = await setup([
+            { key: 'a', text: 'alpha', icon: iconXmarkTag }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'no match' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        expect(() => pageObject.getRenderedCellIcon(0, 0)).toThrowError();
+    });
+
+    it('changing fieldName updates display', async () => {
+        ({ element, connect, disconnect, model } = await setup([
+            { key: 'a', text: 'alpha', icon: iconXmarkTag },
+            { key: 'b', text: 'bravo', icon: iconCheckTag }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'a', field2: 'b' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        model.col1.fieldName = 'field2';
+        await waitForUpdatesAsync();
+
+        expect(
+            pageObject.getRenderedCellIcon(0, 0) instanceof IconCheck
+        ).toBeTrue();
+    });
+
+    it('changing mapping icon updates display', async () => {
```

Comment:
It would be good to have another similar test here that verifies that changing the severity updates the display.